src/schema/bookmarks.ts

```diff
@@ -325,16 +346,41 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
       return { _: true };
     },
     createBookmarkList: async (
-      source,
-      { name }: { name: string },
+      _,
+      { name, icon }: Record<'name' | 'icon', string>,
       ctx: AuthContext,
-    ): Promise<GQLBookmarkList> =>
-      ctx.con.getRepository(BookmarkList).save({
+    ): Promise<GQLBookmarkList> => {
+      const user = await ctx.con.getRepository(User).findOneOrFail({
+        where: { id: ctx.userId },
+        select: ['subscriptionFlags'],
+      });
+      const isPlus = isPlusMember(user.subscriptionFlags?.cycle);
+      const maxFoldersCount = isPlus ? 50 : 1;
+      const userFoldersCount = await ctx.con
+        .getRepository(BookmarkList)
+        .countBy({ userId: ctx.userId });
+
+      if (userFoldersCount >= maxFoldersCount) {
+        if (isPlus) {
+          logger.warn(
+            { listCount: userFoldersCount, userId: ctx.userId },
+            'bookmark folders limit reached',
+          );
+        }
+
+        throw new ForbiddenError(
+          `You have reached the maximum list count (${maxFoldersCount})`,
+        );
+      }
+
+      return ctx.con.getRepository(BookmarkList).save({
         name,
+        icon,
```

Comment:
Missing code to check if its one of the allowed icons?

Reply:
I found a [nice way to check if is an emoji](https://stackoverflow.com/questions/18862256/how-to-detect-emoji-using-javascript), but we need to change tsconfig target. 
Do you think this could be possibile?

Reply:
Nvm used another way and seems working fine ✅